subsys/debug/asan.c

```diff
@@ -16,3 +16,32 @@ const char *__asan_default_options(void)
 #endif
 		"";
 }
+
+const char *__lsan_default_suppressions(void)
+{
+	return
+		/* SDL display driver
+		 *
+		 * The SDL2 library does not clean-up all it resources on exit,
+		 * as such suppress all memory leaks coming from libSDL2 and the
+		 * underlying X11 library
+		 */
+		"leak:libX11\n"
+		"leak:libSDL2\n"
+		;
+}
+
+/* LSAN has a known limitation that if dlcose is called before performing the
+ * leak check "<unknown module>" is reported in the stack traces and these can
+ * not be suppressed, see
+ * https://github.com/google/sanitizers/issues/89 for more info.
+ *
+ * In case that the SDL display driver is used this will result in error
+ * reporting that can not be suppressed, as a workaround implement a NOP version
+ * of dlclose as proposed in the mentioned sanitizer github issue.
+ *
+ */
+int dlclose(void *handler)
```

Comment:
Daaaamm, this is hacky. Having this kind of override always compiled in (just with CONFIG_ASAN) is just not ok.

Reply:
> Daaaamm, this is hacky.

100% agree, but it seem there is no intention to fix this in ASAN so we have to live with it

> Having this kind of override always compiled in (just with CONFIG_ASAN) is just not ok.

Good point; but as you mentioned to put the suppressions under `#ifdef` we could do the same here. My proposal is to add a Kconfig so that the user has the option to enable it or that we can enable it by default when needed (eg. SDL display + ASAN). So majority of the time it will not be in the build as it will only be host specific libraries that use `dlopen` and `dlclose`.

Reply:
If all this new code was under a SDL ifdef, I would be much less fearful of it.
I would even rename this file to "asan_hacks.c" or something like that, to attract people into it if they find themselves trying debugging poltergeists when running with ASAN.